newroutes/authenticatedSites/__tests__/Homepage.spec.js

```diff
@@ -0,0 +1,169 @@
+const express = require("express")
+const request = require("supertest")
+
+const { errorHandler } = require("@middleware/errorHandler")
+const { attachReadRouteHandlerWrapper } = require("@middleware/routeHandler")
+
+const { HomepageRouter } = require("../homepage")
+
+describe("Homepage Router", () => {
+  const mockHomepagePageService = {
+    read: jest.fn(),
+    update: jest.fn(),
+  }
+
+  const router = new HomepageRouter({
+    homepagePageService: mockHomepagePageService,
+  })
+
+  const app = express()
+  app.use(express.json({ limit: "7mb" }))
+  app.use(express.urlencoded({ extended: false }))
+
+  // We can use read route handler here because we don't need to lock the repo
+  app.get(
+    "/:siteName/homepage",
+    attachReadRouteHandlerWrapper(router.readHomepage)
+  )
+  app.post(
+    "/:siteName/homepage",
+    attachReadRouteHandlerWrapper(router.updateHomepage)
+  )
+  app.use(errorHandler)
+
+  const siteName = "test-site"
+  const accessToken = undefined // Can't set request fields - will always be undefined
+  const fileName = "test-file"
+  const mockSha = "12345"
+  const mockContent = "mock-content"
+
+  const reqDetails = { siteName, accessToken }
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  describe("readHomepage", () => {
+    const expectedResponse = {
+      sha: mockSha,
+      content: mockContent,
+    }
+
+    mockHomepagePageService.read.mockResolvedValue(expectedResponse)
+
+    it("retrieves homepage details", async () => {
```

Comment:
could we add 1 extra test case illustrating what happens if our reads fail?

Reply:
Makes sense, https://github.com/isomerpages/isomercms-backend/pull/331/commits/42e416cd091cad47605e8e264d8d0fc788dc3b2c